### PR TITLE
feat: change file filtering

### LINKED ISSUE #121: File filtering
Filter deployment files like:

```shell
$ ls k8s
secret-test.yaml
secret.yaml
common_secret.yaml

$ deploy test
```

`deploy local` should use `secret.yaml` and `common_secret.yaml`  
`deploy test` should use `secret-test.yaml` and `common_secret.yaml` 
`deploy prod` should use `secret.yaml` and `common_secret.yaml`  

I.e if there is a matching "type" file for an environment (`<type>-<env>.yaml`) filter the "raw type file"...

### PR DESCRIPTION
**Pull request by argoyle**
Extracted helper-funcs for filtering files for target.

Closes #121

### CODE REVIEW

**Review by peter-svensson — APPROVED** [review]
Haven't tested but the code looks solid! 
⭐

**Comment by argoyle** [comment]
@peter-svensson I updated the docs a bit. Please have a look at them.

File: www/docs/config/k8s.md [hunk 1]
```diff
@@ -1,45 +1,55 @@
...
+Files with a `.yaml` suffix will
+be [applied](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply) to the Kubernetes cluster.
 
-Files with a `.sh` suffix will be run on the machine executing the [`deploy`](/commands/deploy) command.
-This can be useful to setup secrets/configurations, mostly for local use.
-Note that only `.sh` files matching the `target` will be executed.
+If both a **common** file and a **target-specific** file with the same "basename", i.e. service.yaml and
+service-local.yaml, only the **target-specific** file will be applied to the local target and the **common** file will
+be applied to all other targets.
+
```
Comment (peter-svensson) [inline www/docs/config/k8s.md]:
```suggestion
If both a **common** file and a **target-specific** file with the same "basename", i.e. `service.yaml` and
`service-local.yaml` exist, only the **target-specific** file will be applied for a `target` and the **common** will be applied to all other targets.,.

 to the local target and the **common** file will
be applied to all other targets.

```

**Review by peter-svensson — APPROVED** [review]
LGTM only minor comments on phrasing...

File: www/docs/config/k8s.md (hunk 1 above)
Comment (argoyle) [reply www/docs/config/k8s.md]: Fixed!

**Pull request merged**